experimental/Schemes/singular_locus.jl

```diff
@@ -0,0 +1,147 @@
+export is_equidimensional
+export singular_locus
+export reduced_scheme
+export is_smooth
+
+function singular_locus(X::AbsSpec{<:Ring, <:MPolyQuoLocalizedRing})
+  comp = _singular_locus_with_decomposition(X)
+  if length(comp) == 0 
+    return subscheme(X, one(OO(X)))
+  end
+  R = base_ring(OO(X))
+  return Spec(R, prod([modulus(OO(Y)) for Y in comp]), inverted_set(OO(X)))
```

Comment:
`intersect` is significantly more expensive than taking products. So if people were only interested in the set-theoretic result, they might appreciate the shorter runtime. Maybe this can be resolved by a flag?